external/elasticsearch/src/main/java/com/digitalpebble/stormcrawler/elasticsearch/persistence/AbstractSpout.java

```diff
@@ -164,13 +164,19 @@ public void open(
     protected final boolean addHitToBuffer(SearchHit hit) {
         Map<String, Object> keyValues = hit.getSourceAsMap();
         String url = (String) keyValues.get("url");
+
+        Map<String, List<String>> metadatas = (Map<String, List<String>>) keyValues.get("metadata");
+        addHitInfoToMetadata(metadatas, hit);
+
         // is already being processed - skip it!
         if (beingProcessed.containsKey(url)) {
             return false;
         }
         return buffer.add(url, fromKeyValues(keyValues));
```

Comment:
I would store the value returned by _fromKeyValues(keyValues)_ in a variable 
and have addHitInfoToMetadata() operate on it and become

_addHitInfoToMetadata(Metadata metadata, SearchHit hit)_